SchedulerBase polish, typings, test

erikkemperman opened the pull request:
I am once again trying to separate moderately sized bits off a branch with way too many changes in it...

jcafhe commented on rx/concurrency/schedulerbase.py:
```diff
@@ -6,15 +7,94 @@
+        return default_now()
+
+    @abstractmethod
+    def schedule(self,
```
I'm just curious, is it worth to re-declare `schedule`, `schedule_relative` & `schedule_absolute` as abstract methods while these are already declared in `typing.Scheduler` ?

erikkemperman replied:
Yeah, maybe not strictly necessary... Although it might help to keep the generated docs and `help()` complete.

I guess I included this mainly because of an idea I have that I’m playing around with, which would simplify all of the classes that extend `SchedulerBase` by putting some often-repeated logic (schedule methods being defined in terms of one another) in the base class.

Having these here at this point would make that second step a lot easier / readable.

jcafhe replied:
cool :+1:

jcafhe commented on rx/concurrency/schedulerbase.py:
```diff
@@ -77,57 +157,86 @@ def now(self) -> datetime:
-    def to_timedelta(cls, timespan: typing.AbsoluteOrRelativeTime) -> timedelta:
-        """Converts time value to timedelta"""
+    def to_timedelta(cls, value: typing.AbsoluteOrRelativeTime) -> timedelta:
+        """Converts time value to timedelta. This method handles both absolute
```
Maybe we could specify in the doc that the returned timedelta is relative to POSIX time 0 (January 1, 1970 00:00)

erikkemperman replied:
Yes, good catch, that would be useful to mention! I’ll leave this PR alone for a bit, not to complicate its review, but I’ll add that later!

erikkemperman replied:
I've added a sentence mentioning this, thanks!

erikkemperman commented:
Ugh I just noticed I accidentally included a change in virtualtime scheduler here... It should have been in that other PR.

Sorry about that, I’ve been juggling branches and commits and this fell through. Please ignore that part, I’ll transplant it later.

erikkemperman commented on tests/test_concurrency/test_virtualtimescheduler.py:
```diff
@@ -29,28 +29,38 @@ def action(scheduler, state):
-        ex = 'ex'
+        msg = 'ex'
+        exc = None
         try:
```
Hmm, I did not know about that trick. Thanks! Yes, agree that might be better. But I’ll have to change it in a lot of different places... Maybe I’ll find some time tomorrow.

erikkemperman replied:
I've moved this bit to the other PR, where it belongs. But thanks, I used your suggestion there!

erikkemperman commented:
FYI, I removed the bit which was not supposed to be in this PR, and I added a sentence about `to_timedelta` being relative to the epoch (1970-01-01) as suggested by @jcafhe

erikkemperman commented:
Thanks for review!
